module-4/metascan/metascan.py

```diff
@@ -0,0 +1,91 @@
+import os
+import csv
+import requests
+from bs4 import BeautifulSoup
+from urllib.parse import urljoin, urlparse
+from PyPDF2 import PdfReader
+import argparse
+
+def extract_pdf_links(url):
+    response = requests.get(url)
+    response.raise_for_status()
+    soup = BeautifulSoup(response.text, 'html.parser')
+    pdf_links = [urljoin(url, link.get('href')) for link in soup.find_all('a', href=True) if link.get('href').endswith('.pdf')]
+    return pdf_links
+
+def download_pdf(url, save_path):
+    response = requests.get(url)
+    response.raise_for_status()
```

Comment:
since you are using this exact snippet 2 times, you should maybe abstract it into a function that can be used in various places and will return your HTML content.